cheroot/test/test_ssl.py

```diff
@@ -487,6 +488,28 @@ def test_ssl_env(
             }:
                 assert key in env
 
+    # builtin ssl environment generation may use a loopback socket
+    # ensure no ResourceWarning was raised during the test
+    # NOTE: python 2.7 does not emit ResourceWarning for ssl sockets
+    for warn in recwarn:
+        if not issubclass(warn.category, ResourceWarning):
+            continue
+
+        # the tests can sporadically generate resource warnings
+        # due to timing issues
+        # all of these sporadic warnings appear to be about socket.socket
+        # and have been observed to come from requests connection pool
+        msg = str(warn.message)
+        if 'socket.socket' in msg:
+            pytest.xfail(
+                '\n'.join((
+                    'Sometimes this test fails due to '
+                    'a socket.socket ResourceWarning:',
+                    msg,
+                )),
+            )
+        pytest.fail(reason=msg)  # noqa: PT016
```

Comment:
I take it back. The linter was correct: there's no `reason=` kwarg, it's `msg=`:
```suggestion
        pytest.fail(msg)
```